pkg/windows/calico.go

```diff
@@ -52,147 +137,232 @@ const (
+
+	// Fill all information except for the token
+	calicoKubeConfig := CalicoKubeConfig{
+		Server:               "https://127.0.0.1:6443",
+		CertificateAuthority: filepath.Join("c:\\", c.DataDir, "agent", "server-ca.crt"),
+		Path:                 filepath.Join("c:\\", c.DataDir, "agent", CalicoKubeConfigName),
 	}
 
-	if err := createCNIConfig(cfg); err != nil {
-		return nil, err
+	// Generate the token request
+	req := authv1.TokenRequest{
+		Spec: authv1.TokenRequestSpec{
+			Audiences:         []string{version.Program},
+			ExpirationSeconds: pointer.Int64(60 * 60 * 24 * 365),
+		},
 	}
 
-	logrus.Info("Generating HNS networks, please wait")
-	if err := generateCalicoNetworks(cfg.CalicoConfig.NetworkingBackend); err != nil {
+	// Register the token in the Calico service account
+	client, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
 		return nil, err
 	}
+	serviceAccounts := client.CoreV1().ServiceAccounts(CalicoSystemNamespace)
+	token, err := serviceAccounts.CreateToken(ctx, calicoNode, &req, metav1.CreateOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create token for service account (%s/%s)", CalicoSystemNamespace, calicoNode)
+	}
+
+	calicoKubeConfig.Token = token.Status.Token
 
-	return cfg, nil
+	return &calicoKubeConfig, nil
 }
 
 // Start starts the CNI services on the Windows node.
-func (c *Calico) Start(ctx context.Context, config *CNIConfig) error {
+func (c *Calico) Start(ctx context.Context) error {
 	for {
-		if err := startCalico(ctx, config.CalicoConfig); err != nil {
+		if err := startCalico(ctx, c.CNICfg); err != nil {
 			continue
 		}
 		break
 	}
-	go startFelix(ctx, config.CalicoConfig)
+	go startFelix(ctx, c.CNICfg)
 
 	return nil
 }
 
-// createNodeNameFile creates the node name file required by the CNI.
-func createNodeNameFile(config *CNIConfig, dataDir string) error {
-	return util2.WriteFile(config.CalicoConfig.KubeConfig.Path, config.NodeConfig.AgentConfig.NodeName)
-}
+func (c *Calico) generateCalicoNetworks() error {
+	if err := deleteAllNetworksOnNodeRestart(); err != nil {
+		return err
+	}
 
-// createKubeConfig creates the kube config required by the CNI.
-func createKubeConfig(config *CNIConfig, dataDir string) error {
-	kubeTemplate, err := parseTemplateFromConfig(calicoKubeConfigTemplate, config)
+	mgmt, err := createHnsNetwork(c.CNICfg.Mode, os.Getenv("VXLAN_ADAPTER"))
 	if err != nil {
 		return err
 	}
-	return util2.WriteFile(filepath.Join("c:\\", dataDir, "agent", CalicoKubeConfigName), kubeTemplate)
+
+	if c.CNICfg.Platform == "ec2" || c.CNICfg.Platform == "gce" {
+		logrus.Debugf("recreating metadata route because platform is: %s", c.CNICfg.Platform)
+		if err := setMetaDataServerRoute(mgmt); err != nil {
+			return err
+		}
+	}
+	if c.CNICfg.Mode == "windows-bgp" {
+		// Not supported yet, this does work.
+		_ = wapi.StopService("RemoteAccess")
```

Comment:
Should these be commented out? Are there side effects?

Reply:
Good catch, we are right now hardconding the mode to be vxlan, so this code can be removed